deployment.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: debug-deployment
+  labels:
+    app: debug-app
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: debug-app
+  template:
+    metadata:
+      labels:
+        app: debug-app
+    spec:
+      containers:
+      - name: main-container
+        image: nginx:1.27.1
+        ports:
+        - containerPort: 8080
+        resources:
+          requests:
+            cpu: 100m
+            memory: 128Mi
+          limits:
+            cpu: 200m
+            memory: 256Mi
+       # readinessProbe:
+       #   httpGet:
+       #     path: /healthz
+       #     port: 8080
+       #   initialDelaySeconds: 10
+       #   periodSeconds: 5
+       # livenessProbe:
+       #   httpGet:
+       #     path: /healthz
+       #     port: 8080
+       #   initialDelaySeconds: 15
+       #   periodSeconds: 10
```

Comment:
You can uncomment as the readinessProbe and livenessProbe is required, but fix the `/healthz` endpoint since it's not exist